### PR TITLE
 python310Packages.autobahn: adopt, populate passthru.extras-require 

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 172113` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>37 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>alerta-server: log was empty</li>
<li>cura: log was empty</li>
<li>gns3-gui: log was empty</li>
<li>gns3-server: log was empty</li>
<li>home-assistant-component-tests.sentry: log was empty</li>
<li>moodle-dl: log was empty</li>
<li>moz-phab: log was empty</li>
<li>napalm: log was empty</li>
<li>netbox: log was empty</li>
<li>octoprint: log was empty</li>
<li>odoo: log was empty</li>
<li>openai: log was empty</li>
<li>privacyidea: log was empty</li>
<li>python310Packages.augmax: log was empty</li>
<li>python310Packages.einops: log was empty</li>
<li>python310Packages.ldaptor: log was empty</li>
<li>python310Packages.mailman: log was empty</li>
<li>python310Packages.mailman-hyperkitty: log was empty</li>
<li>python310Packages.napalm: log was empty</li>
<li>python310Packages.napalm-hp-procurve: log was empty</li>
<li>python310Packages.privacyidea-ldap-proxy: log was empty</li>
<li>python310Packages.trytond: log was empty</li>
<li>python39Packages.django-rq: log was empty</li>
<li>python39Packages.ldaptor: log was empty</li>
<li>python39Packages.napalm: log was empty</li>
<li>python39Packages.napalm-hp-procurve: log was empty</li>
<li>python39Packages.privacyidea-ldap-proxy: log was empty</li>
<li>python39Packages.sentry-sdk: log was empty</li>
<li>python39Packages.spacy: log was empty</li>
<li>python39Packages.spacy-loggers: log was empty</li>
<li>python39Packages.spacy-transformers: log was empty</li>
<li>python39Packages.textacy: log was empty</li>
<li>trytond: log was empty</li>
<li>python39Packages.wandb: log was empty</li>
<li>skytemple: log was empty</li>
<li>tribler: log was empty</li>
<li>zeroc-ice-cpp11: log was empty</li>
  </ul>
</details>
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>(python39Packages.openai): log was empty</li>
<li>(python39Packages.trytond): log was empty</li>
<li>wolfram-notebook: log was empty</li>
  </ul>
</details>
<details>
<summary>
142 packages built:
</summary>
<ul>
<li>buildbot (python39Packages.buildbot)</li>
<li>buildbot-full (python39Packages.buildbot-full)</li>
<li>buildbot-ui (python39Packages.buildbot-ui)</li>
<li>buildbot-worker (python39Packages.buildbot-worker)</li>
<li>chia</li>
<li>datadog-agent</li>
<li>datadog-process-agent</li>
<li>devpi-client</li>
<li>devpi-server</li>
<li>dyndnsc</li>
<li>etebase-server (python39Packages.etebase-server)</li>
<li>etesync-dav</li>
<li>gnome-keysign</li>
<li>gnome-secrets</li>
<li>grab-site</li>
<li>home-assistant-component-tests.roomba</li>
<li>ihaskell</li>
<li>jupyter</li>
<li>magic-wormhole (python39Packages.magic-wormhole)</li>
<li>mailman</li>
<li>mitmproxy (python39Packages.mitmproxy)</li>
<li>murmur</li>
<li>paperless-ngx</li>
<li>passExtensions.pass-import</li>
<li>pgadmin4</li>
<li>python310Packages.Nikola</li>
<li>python310Packages.amqtt</li>
<li>python310Packages.ansible-kernel</li>
<li>python310Packages.argon2-cffi</li>
<li>python310Packages.autobahn</li>
<li>python310Packages.batchspawner</li>
<li>python310Packages.buildbot</li>
<li>python310Packages.buildbot-full</li>
<li>python310Packages.buildbot-ui</li>
<li>python310Packages.buildbot-worker</li>
<li>python310Packages.channels</li>
<li>python310Packages.channels-redis</li>
<li>python310Packages.ciscoconfparse</li>
<li>python310Packages.daphne</li>
<li>python310Packages.dependency-injector</li>
<li>python310Packages.dockerspawner</li>
<li>python310Packages.etebase-server</li>
<li>python310Packages.fastapi</li>
<li>python310Packages.flask-security-too</li>
<li>python310Packages.jupyter</li>
<li>python310Packages.jupyter-lsp</li>
<li>python310Packages.jupyter-server-mathjax</li>
<li>python310Packages.jupyter_server</li>
<li>python310Packages.jupyterhub</li>
<li>python310Packages.jupyterhub-ldapauthenticator</li>
<li>python310Packages.jupyterhub-systemdspawner</li>
<li>python310Packages.jupyterhub-tmpauthenticator</li>
<li>python310Packages.jupyterlab</li>
<li>python310Packages.jupyterlab-git</li>
<li>python310Packages.jupyterlab-lsp</li>
<li>python310Packages.jupyterlab_launcher</li>
<li>python310Packages.jupyterlab_server</li>
<li>python310Packages.jupytext</li>
<li>python310Packages.keyrings-cryptfile</li>
<li>python310Packages.labgrid</li>
<li>python310Packages.magic-wormhole</li>
<li>python310Packages.magic-wormhole-mailbox-server</li>
<li>python310Packages.magic-wormhole-transit-relay</li>
<li>python310Packages.mitmproxy</li>
<li>python310Packages.nbclassic</li>
<li>python310Packages.nbdime</li>
<li>python310Packages.nix-kernel</li>
<li>python310Packages.notebook</li>
<li>python310Packages.notedown</li>
<li>python310Packages.oauthenticator</li>
<li>python310Packages.ormar</li>
<li>python310Packages.passlib</li>
<li>python310Packages.pg8000</li>
<li>python310Packages.pykeepass</li>
<li>python310Packages.pypiserver</li>
<li>python310Packages.roombapy</li>
<li>python310Packages.slowapi</li>
<li>python310Packages.testing-postgresql</li>
<li>python310Packages.vega</li>
<li>python310Packages.volvooncall</li>
<li>python39Packages.Nikola</li>
<li>python39Packages.amqtt</li>
<li>python39Packages.ansible-kernel</li>
<li>python39Packages.argon2-cffi</li>
<li>python39Packages.augmax</li>
<li>python39Packages.autobahn</li>
<li>python39Packages.batchspawner</li>
<li>python39Packages.channels</li>
<li>python39Packages.channels-redis</li>
<li>python39Packages.ciscoconfparse</li>
<li>python39Packages.daphne</li>
<li>python39Packages.dependency-injector</li>
<li>python39Packages.dockerspawner</li>
<li>python39Packages.einops</li>
<li>python39Packages.elegy</li>
<li>python39Packages.fastapi</li>
<li>python39Packages.flask-security-too</li>
<li>python39Packages.json-logging</li>
<li>python39Packages.jupyter</li>
<li>python39Packages.jupyter-lsp</li>
<li>python39Packages.jupyter-server-mathjax</li>
<li>python39Packages.jupyter_server</li>
<li>python39Packages.jupyterhub</li>
<li>python39Packages.jupyterhub-ldapauthenticator</li>
<li>python39Packages.jupyterhub-systemdspawner</li>
<li>python39Packages.jupyterhub-tmpauthenticator</li>
<li>python39Packages.jupyterlab</li>
<li>python39Packages.jupyterlab-git</li>
<li>python39Packages.jupyterlab-lsp</li>
<li>python39Packages.jupyterlab_launcher</li>
<li>python39Packages.jupyterlab_server</li>
<li>python39Packages.jupytext</li>
<li>python39Packages.keyrings-cryptfile</li>
<li>python39Packages.labgrid</li>
<li>python39Packages.magic-wormhole-mailbox-server</li>
<li>python39Packages.magic-wormhole-transit-relay</li>
<li>python39Packages.mailman</li>
<li>python39Packages.mailman-hyperkitty</li>
<li>python39Packages.nbclassic</li>
<li>python39Packages.nbdime</li>
<li>python39Packages.nix-kernel</li>
<li>python39Packages.notebook</li>
<li>python39Packages.notedown</li>
<li>python39Packages.oauthenticator</li>
<li>python39Packages.ormar</li>
<li>python39Packages.passlib</li>
<li>python39Packages.pg8000</li>
<li>python39Packages.pykeepass</li>
<li>python39Packages.pypiserver</li>
<li>python39Packages.roombapy</li>
<li>python39Packages.slowapi</li>
<li>python39Packages.testing-postgresql</li>
<li>python39Packages.treex</li>
<li>python39Packages.vega</li>
<li>python39Packages.volvooncall</li>
<li>qutebrowser</li>
<li>radicale (radicale3)</li>
<li>radicale2</li>
<li>soco-cli</li>
<li>tahoe-lafs</li>
<li>theharvester</li>
<li>zeroc-ice</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
LGTM and ready to be merged.

**Review by asymmetric — APPROVED** [review]
Tested `magic-wormhole` and it works - thanks!

File: pkgs/development/python-modules/autobahn/default.nix [hunk 1]
```diff
@@ -1,25 +1,49 @@
...
+, attrs
+, argon2-cffi
+, base58
 , cbor2
 , cffi
+, click
 , cryptography
-, fetchPypi
+, ecdsa
+  # , eth-abi
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/python-modules/autobahn/default.nix:12]: They are not packaged yet.

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: pkgs/development/python-modules/autobahn/default.nix [hunk 2]
```diff
@@ -74,10 +89,23 @@ buildPythonPackage rec {
     "autobahn"
   ];
 
+  passthru.extras-require = rec {
+    all = accelerate ++ compress ++ encryption ++ nvx ++ serialization ++ scram ++ twisted ++ ui ++ xbr;
```
[the remark this answers is not in the document]
Comment (asymmetric) [reply pkgs/development/python-modules/autobahn/default.nix:93]: OK, but they're declared in another order just below, so there's a tiny inconsistency. Nothing important, just a nit.

**Pull request merged**